snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -646,7 +641,7 @@ def _generate_part_env(self, step: steps.Step) -> str:
             for k, v in plugin_environment.items():
                 print(f'export {k}="{v}"', file=run_environment)
             print("## User Environment", file=run_environment)
-            for env in user_build_environment:
+            for env in self.build_environment:
```

Comment:
or (I prefer this even if it is more code) combined with the change some lines of code prior

```suggestion
            for env in user_build_environment:
```

Reply:
Done.